### PR TITLE
ReactTable HOC: advancedExpandTable

### PR DESCRIPTION
**Pull request by mayank23**
Hi! 😄 

My team uses React-Table heavily! Recently we had to support being able to expand/collapse the subcomponent from any Cell in the row. Hence, we leveraged ReactTable's HOC capability to implement this. 

Description:

**AvancedExpandTableHOC for ReactTable**
  - HOC which allows any Cell in the row to toggle the row's
  SubComponent. Also allows the SubComponent to toggle itself.
  - Expand functions available to any SubComponent or Column Cell: 
     - toggleRowSubComponent
     - showRowSubComponent
     - hideRowSubComponent
 - Each Column Renderer (E.g. Cell ) gets the expand functions in its props and Each SubComponent gets the expand functions in its props
 - Expand functions takes the `rowInfo` given to each Column Renderer and SubComponent already by ReactTable.
- accepts an `onExpandedChange` callback to be called whenever the table expanded state changes
- currently only supports 1 level of nesting for the sub component.

**Here is a working example:** https://codesandbox.io/s/9v5788y64 *(same one included in the PR also)*

In this example, you can toggle, show, hide the subcomponent from any Cell in its corresponding row, and also close the subcomponent from inside the subcomponent itself.

I've added the main code to the HOC folder, updated the HOC readme, and added the codesandbox example to the `examples` folder

Please let me know if this is fine to add in and if I should make any changes 😄

Thanks a lot!

**BTW**, Travis is failing due to the linter having errors on the `foldable` table HOC

### CODE REVIEW

**Comment by mayank23** [comment]
Hi, @tannerlinsley does this look good? 😄

**Comment by mayank23** [comment]
Hi @tannerlinsley , sorry to bother, would could you check this PR when you get some time? Thank you 😄

**Comment by tannerlinsley** [comment]
This feature/issue has been tagged as one that will likely be fixed or improved in the next major release of React-Table. This release is expected to be completed in the next 6 - 8 months. Please see #1143 for more detail. If you believe this feature can be easily fixed, we invite you to fork the project, attempt the fix, and use it until the latest version is available. If the fix is simple enough, we will also consider it as a PR to the current version.

File: src/hoc/advancedExpandTable/index.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import set from 'lodash.set'
```
Comment (simonv3) [inline src/hoc/advancedExpandTable/index.js]: Should these have been added to the package.json @mayank23 ?

Comment (cwackerfuss) [reply src/hoc/advancedExpandTable/index.js]: Likely. I get an error when using this HOC:
```
Module not found: Can't resolve 'lodash.get' in '/Users/cwack/code/branchweb/node_modules/react-table/lib/hoc/advancedExpandTable'
```

Comment (mayank23) [reply src/hoc/advancedExpandTable/index.js]: Hey @cwackerfuss I don't think the maintainer is going to accept any PRs soon, so it's probably best to fork it and add it to the dependencies. Or you can just copy over the file and add it to your project. I'll see If I can get a standalone repo up soon too. Thanks 😄

**Comment by brandondurham** [comment]
Has this been ported over to use hooks, by chance? 😬

**Pull request merged**